File: threatguard-frontend/src/types/threat.ts
export type IocType = "ip" | "domain" | "url" | "hash";

export type Classification = "LOW" | "MEDIUM" | "HIGH" | "CRITICAL";

export interface SourceResult {
  source: string;
  supported: boolean;
  error?: string;
  abuse_score?: number;
  total_reports?: number;
  country?: string;
  isp?: string;
  is_tor?: boolean;
  pulse_count?: number;
  reputation?: number;
  asn?: string;
  query_status?: string;
  threat?: string;
  url_status?: string;
  tags?: string[];
  file_type?: string;
  signature?: string;
}

export interface ThreatResponse {
  ioc_type: IocType;
  ioc_value: string;
  risk_score: number;
  classification: Classification;
  from_cache: boolean;
  sources: SourceResult[];
}

File: threat-service/app/schemas/threat.py
from pydantic import BaseModel
from typing import Any

class SourceResult(BaseModel):
    source: str
    supported: bool = True
    error: str | None = None

class ThreatResponse(BaseModel):
    ioc_type: str
    ioc_value: str
    risk_score: int
    classification: str  # LOW, MEDIUM, HIGH, CRITICAL
    sources: list[dict[str, Any]]
    from_cache: bool